File: Data Structures & Algorithms/lowest-common-ancestor-of-a-binary-tree-iii/submission-0.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None
        self.parent = None
"""

class Solution:
    def lowestCommonAncestor(self, p: 'Node', q: 'Node') -> 'Node':
        pAnc, qAnc=deque(),deque()
        while p!=None:
            pAnc.append(p)
            p=p.parent
        
        while q!=None:
            qAnc.append(q)
            q=q.parent

        for i in pAnc:
            for j in qAnc:
                if i==j:
                    return i
        

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int cnt=1;
    int arr[]; 
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> arr= new ArrayList<>();
        dfs(root,arr);
        return arr.get(k-1);
    }

    void dfs(TreeNode root, List<Integer> arr){
        if(root==null) return;
        if(root.left!=null) dfs(root.left,arr);
        arr.add(root.val);    
        if(root.right!=null) dfs(root.right,arr);
    }
}
